heap.c

```diff
@@ -100,6 +110,20 @@ siftup(struct heap *h, size_t pos)
 }
 
 void
+heap_heapify(struct heap *h)
+{
+	for (ssize_t i=ptrvec_size(h->vec)/2-1; i>=0; i--) {
```

Comment:
however, a size_t/2 can never overflow a ssize_t.  I can rewrite as (size_t i=size/2; i-- >= 0; ) if you really want.